client/components/suggestions/index.jsx

```diff
@@ -0,0 +1,248 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import {
+	noop,
+	pick,
+	pickBy
+}	from 'lodash';
+
+const Suggestions = React.createClass( {
+
+	propTypes: {
+		suggest: React.PropTypes.func,
+		terms: React.PropTypes.object,
+		input: React.PropTypes.string
+	},
+
+	getDefaultProps: function() {
+		return {
+			welcomeSign: noop,
+			suggest: noop,
+			terms: {},
+			input: '',
+		};
+	},
+
+	getInitialState: function() {
+		return {
+			taxonomySuggestionsArray: [],
+			suggestionPosition: -1,
+			suggestions: {},
+			filterTerm: ''
+		};
+	},
+
+	componentWillMount: function() {
+		const suggestions = this.narrowDown( this.props.input );
+		this.setState( {
+			suggestions: suggestions,
+			taxonomySuggestionsArray: this.createTaxonomySuggestionsArray( suggestions ),
+			currentSuggestion: '',
+		} );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.input !== this.props.input ) {
+			const suggestions = this.narrowDown( nextProps.input );
+			this.setState( {
+				suggestions: suggestions,
+				taxonomySuggestionsArray: this.createTaxonomySuggestionsArray( suggestions ),
+				suggestionPosition: -1,
+				currentSuggestion: '',
+			} );
+		}
+	},
+
+	countSuggestions: function() {
+		return this.state.taxonomySuggestionsArray.length;
+	},
+
+	getSuggestionForPosition: function( position ) {
+		return this.state.taxonomySuggestionsArray[ position ];
+	},
+
+	getPositionForSuggestion: function( suggestion ) {
+		return this.state.taxonomySuggestionsArray.indexOf( suggestion );
+	},
+
+	incPosition: function() {
+		const position = ( this.state.suggestionPosition + 1 ) % this.countSuggestions();
+		this.setState( {
+			suggestionPosition: position,
+			currentSuggestion: this.getSuggestionForPosition( position ),
+		} );
+	},
+
+	decPosition: function() {
+		const position = ( this.state.suggestionPosition - 1 ) % this.countSuggestions();
+		this.setState( {
+			suggestionPosition: position,
+			currentSuggestion: this.getSuggestionForPosition( position )
+		} );
+	},
+
+	handleKeyEvent: function( event ) {
+		switch ( event.key ) {
+			case 'ArrowDown' :
+				this.incPosition();
+				event.preventDefault();
+				break;
+			case 'ArrowUp' :
+				this.decPosition();
+				event.preventDefault();
+				break;
+			case 'Enter' :
+				if ( this.state.suggestionPosition !== -1 ) {
+					this.props.suggest( this.state.currentSuggestion );
+				}
+				break;
+		}
+	},
+
+	onMouseDown: function( event ) {
+		this.props.suggest( event.target.textContent );
+	},
+
+	onMouseOver: function( event ) {
+		this.setState( {
+			suggestionPosition: this.getPositionForSuggestion( event.target.textContent ),
+			currentSuggestion: event.target.textContent,
+		} );
+	},
+
+	removeEmptySuggestions: function( suggestions ) {
+		const hasValues = x => x.length > 0;
```

Comment:
You could use lodash's [`isEmpty`](https://lodash.com/docs/4.16.4#isEmpty) here, I think.